Neos.Flow/Classes/Http/Uri.php

```diff
@@ -354,11 +356,276 @@ public function getFragment()
      */
     public function setFragment($fragment)
     {
-        if (preg_match(self::PATTERN_MATCH_FRAGMENT, $fragment) === 1) {
-            $this->fragment = $fragment;
-        } else {
+        if (preg_match(self::PATTERN_MATCH_FRAGMENT, $fragment) !== 1) {
             throw new \InvalidArgumentException('"' . $fragment . '" is not valid fragment as part of a URI.', 1184071252);
         }
+
+        $this->fragment = $fragment;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getHostAndOptionalPort()
+    {
+        $hostAndPort = (string)$this->host;
```

Comment:
Probably not, but I have gotten inspiration from a bunch of other implementations that did it. Right now I must admit I no longer know exactly why, but I would for now rather want to leave it in.